rest_framework/serializers.py

```diff
@@ -801,9 +801,8 @@ def raise_errors_on_nested_writes(method_name, serializer, validated_data):
     #     ...
     #     address = serializer.CharField('profile.address')
     assert not any(
-        '.' in field.source and
-        (key in validated_data) and
-        isinstance(validated_data[key], (list, dict))
```

Comment:
This check is implicitly included in `len(field.source_attrs) > 1` due to `set_value` creating a nested dict when there is a dotted source.